---
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# singscore <img src="man/figures/logo.png" align="right"  height="140" width="120" alt="logo"/>

## Overview

'singscore' is an R/Bioconductor package which implements the simple single-sample gene-set (or gene-signature) scoring method proposed by Foroutan *et al.* (2018). It uses rank-based statistics to analyze each sample's gene expression profile and scores the expression activities of gene sets at a single-sample level.

We have written up a **new** workflow package demonstrating application of singscore to infer mutation status in the TCGA acute myeloid leukemia cohort. Refer to the published workflow at <https://f1000research.com/articles/8-776/v2>. It is also available as a R/Bioconductor workflow package `SingscoreAMLMutations`.

## Getting Started

These instructions will get you to install the package up and running on your local machine. If you experience any issues, please raise a GitHub issue at <https://github.com/DavisLaboratory/singscore/issues>.

```
# build_vignettes = TRUE to build vignettes upon installation
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("singscore", version = "3.8")
```

## Documentation

The package comes with a vignette showing how the different functions in the package can be used to perform a gene-set enrichment analysis on a single sample level. Pre-built vignettes can be accessed via [Bioconductor](https://bioconductor.org/packages/release/bioc/vignettes/singscore/inst/doc/singscore.html) or [the GitHub IO page](https://davislaboratory.github.io/singscore/articles/singscore.html).

## References
Foroutan, Momeneh, Dharmesh D Bhuva, Ruqian Lyu, Kristy Horan, Joseph Cursons, and Melissa J Davis. 2018. “Single Sample Scoring of Molecular Phenotypes.” BMC Bioinformatics 19 (1). BioMed Central: 404. doi: [10.1186/s12859-018-2435-4](https://doi.org/10.1186/s12859-018-2435-4).
