<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Rank genes by the gene expression intensities — rankGenes • singscore</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Rank genes by the gene expression intensities — rankGenes" />
<meta property="og:description" content="The rankGenes function is a generic function that can deal
with mutilple types of inputs. Given a matrix of gene expression that has
samples in columns, genes in rows, and values being gene expression
intensity,rankGenes ranks gene expression intensities in each sample.
It can also work with S4 objects that have gene expression matrix as a
component (i.e ExpressionSet, DGEList,SummarizedExperiment). It calls the
rank function in the base package which ranks the gene expression
matrix by its absolute expression level. If the input is S4 object of
DGEList, ExpressionSet, or SummarizedExperiment, it will extract the
gene expression matrix from the object and rank the genes. The default
'tiesMethod' is set to 'min'." />
<meta property="og:image" content="/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singscore</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.7.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/singscore.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/DavisLaboratory/singscore">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Rank genes by the gene expression intensities</h1>
    <small class="dont-index">Source: <a href='https://github.com/DavisLaboratory/singscore/blob/master/R/rankGenesGeneric.R'><code>R/rankGenesGeneric.R</code></a></small>
    <div class="hidden name"><code>rankGenes.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The <code>rankGenes</code> function is a generic function that can deal
with mutilple types of inputs. Given a matrix of gene expression that has
samples in columns, genes in rows, and values being gene expression
intensity,<code>rankGenes</code> ranks gene expression intensities in each sample.</p>
<p>It can also work with S4 objects that have gene expression matrix as a
component (i.e ExpressionSet, DGEList,SummarizedExperiment). It calls the
<code>rank</code> function in the base package which ranks the gene expression
matrix by its absolute expression level. If the input is S4 object of
<code>DGEList, ExpressionSet, or SummarizedExperiment</code>, it will extract the
gene expression matrix from the object and rank the genes. The default
'tiesMethod' is set to 'min'.</p>
    </div>

    <pre class="usage"><span class='fu'>rankGenes</span>(<span class='no'>expreMatrix</span>, <span class='kw'>tiesMethod</span> <span class='kw'>=</span> <span class='st'>"min"</span>, <span class='kw'>stableGenes</span> <span class='kw'>=</span> <span class='kw'>NULL</span>)

<span class='co'># S4 method for matrix</span>
<span class='fu'>rankGenes</span>(<span class='no'>expreMatrix</span>, <span class='kw'>tiesMethod</span> <span class='kw'>=</span> <span class='st'>"min"</span>, <span class='kw'>stableGenes</span> <span class='kw'>=</span> <span class='kw'>NULL</span>)

<span class='co'># S4 method for data.frame</span>
<span class='fu'>rankGenes</span>(<span class='no'>expreMatrix</span>, <span class='kw'>tiesMethod</span> <span class='kw'>=</span> <span class='st'>"min"</span>, <span class='kw'>stableGenes</span> <span class='kw'>=</span> <span class='kw'>NULL</span>)

<span class='co'># S4 method for DGEList</span>
<span class='fu'>rankGenes</span>(<span class='no'>expreMatrix</span>, <span class='kw'>tiesMethod</span> <span class='kw'>=</span> <span class='st'>"min"</span>, <span class='kw'>stableGenes</span> <span class='kw'>=</span> <span class='kw'>NULL</span>)

<span class='co'># S4 method for ExpressionSet</span>
<span class='fu'>rankGenes</span>(<span class='no'>expreMatrix</span>, <span class='kw'>tiesMethod</span> <span class='kw'>=</span> <span class='st'>"min"</span>, <span class='kw'>stableGenes</span> <span class='kw'>=</span> <span class='kw'>NULL</span>)

<span class='co'># S4 method for SummarizedExperiment</span>
<span class='fu'>rankGenes</span>(<span class='no'>expreMatrix</span>, <span class='kw'>tiesMethod</span> <span class='kw'>=</span> <span class='st'>"min"</span>, <span class='kw'>stableGenes</span> <span class='kw'>=</span> <span class='kw'>NULL</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>expreMatrix</th>
      <td><p>matrix, data.frame, ExpressionSet, DGEList or
SummarizedExperiment storing gene expression measurements</p></td>
    </tr>
    <tr>
      <th>tiesMethod</th>
      <td><p>character, indicating what method to use when dealing with
ties</p></td>
    </tr>
    <tr>
      <th>stableGenes</th>
      <td><p>character, containing a list of stable genes to be used to
rank genes using expression of stable genes. This is required when using the
stable genes dependent version of singscore (see details in <code>simpleScore</code>).
Stable genes for solid cancers (carcinomas) and blood transcriptomes can be
obtained using the <code>getStableGenes</code> function</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>The ranked gene expression matrix that has samples in columns and
genes in rows. Unit normalised ranks are returned if data is ranked using
stable genes</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='getStableGenes.html'>getStableGenes</a></code>, <code><a href='simpleScore.html'>simpleScore</a></code>,
<code><a href='https://rdrr.io/r/base/rank.html'>rank</a></code>, "ExpressionSet",
"SummarizedExperiment",
"DGEList"</p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'>rankGenes</span>(<span class='no'>toy_expr_se</span>) <span class='co'># toy_expr_se is a gene expression dataset</span></div><div class='output co'>#&gt;    D_Ctrl_R1 D_TGFb_R1
#&gt; 2          2         2
#&gt; 9         13        12
#&gt; 10         4         5
#&gt; 12        10         8
#&gt; 13         8         7
#&gt; 14        18        18
#&gt; 15         3         1
#&gt; 16        20        20
#&gt; 18         6        10
#&gt; 19        14        14
#&gt; 20         9         9
#&gt; 21        11        11
#&gt; 22        17        16
#&gt; 23        19        19
#&gt; 24         5         4
#&gt; 25        16        17
#&gt; 26         1         3
#&gt; 27        12        13
#&gt; 28         7         6
#&gt; 29        15        15
#&gt; attr(,"stable")
#&gt; [1] FALSE</div><div class='input'>
<span class='co'># ExpressionSet object</span>
<span class='no'>emat</span> <span class='kw'>&lt;-</span> <span class='kw pkg'>SummarizedExperiment</span><span class='kw ns'>::</span><span class='fu'><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assay</a></span>(<span class='no'>toy_expr_se</span>)
<span class='no'>e</span> <span class='kw'>&lt;-</span> <span class='kw pkg'>Biobase</span><span class='kw ns'>::</span><span class='fu'><a href='https://rdrr.io/pkg/Biobase/man/class.ExpressionSet.html'>ExpressionSet</a></span>(<span class='kw'>assayData</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/matrix.html'>as.matrix</a></span>(<span class='no'>emat</span>))
<span class='fu'>rankGenes</span>(<span class='no'>e</span>)</div><div class='output co'>#&gt;    D_Ctrl_R1 D_TGFb_R1
#&gt; 2          2         2
#&gt; 9         13        12
#&gt; 10         4         5
#&gt; 12        10         8
#&gt; 13         8         7
#&gt; 14        18        18
#&gt; 15         3         1
#&gt; 16        20        20
#&gt; 18         6        10
#&gt; 19        14        14
#&gt; 20         9         9
#&gt; 21        11        11
#&gt; 22        17        16
#&gt; 23        19        19
#&gt; 24         5         4
#&gt; 25        16        17
#&gt; 26         1         3
#&gt; 27        12        13
#&gt; 28         7         6
#&gt; 29        15        15
#&gt; attr(,"stable")
#&gt; [1] FALSE</div><div class='input'>
<span class='co'>#scoring using the stable version of singscore</span>
<span class='fu'>rankGenes</span>(<span class='no'>e</span>, <span class='kw'>stableGenes</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>'2'</span>, <span class='st'>'20'</span>, <span class='st'>'25'</span>))</div><div class='output co'>#&gt;    D_Ctrl_R1 D_TGFb_R1
#&gt; 2       0.25      0.25
#&gt; 9       0.75      0.75
#&gt; 10      0.50      0.50
#&gt; 12      0.75      0.50
#&gt; 13      0.50      0.50
#&gt; 14      1.00      1.00
#&gt; 15      0.50      0.25
#&gt; 16      1.00      1.00
#&gt; 18      0.50      0.75
#&gt; 19      0.75      0.75
#&gt; 20      0.50      0.50
#&gt; 21      0.75      0.75
#&gt; 22      1.00      0.75
#&gt; 23      1.00      1.00
#&gt; 24      0.50      0.50
#&gt; 25      0.75      0.75
#&gt; 26      0.25      0.50
#&gt; 27      0.75      0.75
#&gt; 28      0.50      0.50
#&gt; 29      0.75      0.75
#&gt; attr(,"stable")
#&gt; [1] TRUE</div><div class='input'>
<span class='kw'>if</span> (<span class='fl'>FALSE</span>) {
<span class='co'>#for real cancer or blood datasets, use getStableGenes()</span>
<span class='fu'>rankGenes</span>(<span class='no'>cancer_expr</span>, <span class='kw'>stableGenes</span> <span class='kw'>=</span> <span class='fu'><a href='getStableGenes.html'>getStableGenes</a></span>(<span class='fl'>5</span>))
<span class='fu'>rankGenes</span>(<span class='no'>blood_expr</span>, <span class='kw'>stableGenes</span> <span class='kw'>=</span> <span class='fu'><a href='getStableGenes.html'>getStableGenes</a></span>(<span class='fl'>5</span>, <span class='kw'>type</span> <span class='kw'>=</span> <span class='st'>'blood'</span>))
}</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Ruqian Lyu, Momeneh Foroutan, Dharmesh D. Bhuva.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


