<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>single-sample gene-set scoring method — simpleScore • singscore</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="single-sample gene-set scoring method — simpleScore" />
<meta property="og:description" content="This function computes 'singscores' using an unmodified
ranked gene expression matrix obtained from the rankGenes() function and a
gene set or a pair of up-regulated and down-regulated gene sets. It returns
a data.frame of scores and dispersions for each sample. The gene sets can be
in vector format or as GeneSet objects (from GSEABase packages). If samples
need to be scored against a single gene set, the upSet argument
should be used to pass the gene set while the downSet argument is set
to NULL. This setting is ideal for gene sets representing gene
ontologies where the nature of the genes is unknown (up- or down-regulated)." />
<meta property="og:image" content="/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">singscore</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.7.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/singscore.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/DavisLaboratory/singscore">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>single-sample gene-set scoring method</h1>
    <small class="dont-index">Source: <a href='https://github.com/DavisLaboratory/singscore/blob/master/R/simpleScoreGeneric.R'><code>R/simpleScoreGeneric.R</code></a></small>
    <div class="hidden name"><code>simpleScore.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function computes 'singscores' using an <strong>unmodified</strong>
ranked gene expression matrix obtained from the <code><a href='rankGenes.html'>rankGenes()</a></code> function and a
gene set or a pair of up-regulated and down-regulated gene sets. It returns
a data.frame of scores and dispersions for each sample. The gene sets can be
in vector format or as GeneSet objects (from GSEABase packages). If samples
need to be scored against a single gene set, the <code>upSet</code> argument
should be used to pass the gene set while the <code>downSet</code> argument is set
to <code>NULL</code>. This setting is ideal for gene sets representing gene
ontologies where the nature of the genes is unknown (up- or down-regulated).</p>
    </div>

    <pre class="usage"><span class='fu'>simpleScore</span>(
  <span class='no'>rankData</span>,
  <span class='no'>upSet</span>,
  <span class='kw'>downSet</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>subSamples</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>centerScore</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>dispersionFun</span> <span class='kw'>=</span> <span class='no'>mad</span>,
  <span class='kw'>knownDirection</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>
)

<span class='co'># S4 method for matrix,vector,missing</span>
<span class='fu'>simpleScore</span>(
  <span class='no'>rankData</span>,
  <span class='no'>upSet</span>,
  <span class='kw'>downSet</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>subSamples</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>centerScore</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>dispersionFun</span> <span class='kw'>=</span> <span class='no'>mad</span>,
  <span class='kw'>knownDirection</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>
)

<span class='co'># S4 method for matrix,GeneSet,missing</span>
<span class='fu'>simpleScore</span>(
  <span class='no'>rankData</span>,
  <span class='no'>upSet</span>,
  <span class='kw'>downSet</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>subSamples</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>centerScore</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>dispersionFun</span> <span class='kw'>=</span> <span class='no'>mad</span>,
  <span class='kw'>knownDirection</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>
)

<span class='co'># S4 method for matrix,vector,vector</span>
<span class='fu'>simpleScore</span>(
  <span class='no'>rankData</span>,
  <span class='no'>upSet</span>,
  <span class='kw'>downSet</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>subSamples</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>centerScore</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>dispersionFun</span> <span class='kw'>=</span> <span class='no'>mad</span>,
  <span class='kw'>knownDirection</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>
)

<span class='co'># S4 method for matrix,GeneSet,GeneSet</span>
<span class='fu'>simpleScore</span>(
  <span class='no'>rankData</span>,
  <span class='no'>upSet</span>,
  <span class='kw'>downSet</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>subSamples</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>centerScore</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>dispersionFun</span> <span class='kw'>=</span> <span class='no'>mad</span>,
  <span class='kw'>knownDirection</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>rankData</th>
      <td><p>A matrix object, ranked gene expression matrix data generated
using the <code><a href='rankGenes.html'>rankGenes()</a></code> function (make sure this matrix is not modified, see
details)</p></td>
    </tr>
    <tr>
      <th>upSet</th>
      <td><p>A GeneSet object or character vector of gene IDs of up-regulated
gene set or a gene set where the nature of genes is not known</p></td>
    </tr>
    <tr>
      <th>downSet</th>
      <td><p>A GeneSet object or character vector of gene IDs of
down-regulated gene set or NULL where only a single gene set is provided</p></td>
    </tr>
    <tr>
      <th>subSamples</th>
      <td><p>A vector of sample labels/indices that will be used to
subset the rankData matrix. All samples will be scored if not provided</p></td>
    </tr>
    <tr>
      <th>centerScore</th>
      <td><p>A Boolean, specifying whether scores should be centered
around 0, default as TRUE. Note: scores never centered if <code>knownDirection = FALSE</code></p></td>
    </tr>
    <tr>
      <th>dispersionFun</th>
      <td><p>A function, dispersion function with default being <code>mad</code></p></td>
    </tr>
    <tr>
      <th>knownDirection</th>
      <td><p>A boolean, determining whether the gene set should be
considered to be directional or not. A gene set is directional if the type
of genes in it are known i.e. up- or down-regulated. This should be set to
TRUE if the gene set is composed of both up- AND down-regulated genes.
Defaults to TRUE. This parameter becomes irrelevant when both upSet(Colc)
and downSet(Colc) are provided.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A data.frame consists of singscores and dispersions for all samples</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Signature scores can be computed using transcriptome-wide
measurements or using a smaller set of measuremnts. If ranks are computed
using the default invocation of <code>rankgenes</code>, the former method is applied
where the rank of each gene in the signature is computed relative to all
other genes in the dataset. Accuracy of this approximation of the relative
expression of a gene will be improved if all or most transctripts are
measured in the experiment. This was the approach proposed in the original
manucript of singscore (Foroutan M, Bhuva DD, et al 2018).</p>
<p>If instead a selected panel of genes is measured (such as from nanostring or
RT-qPCR), a different rank approximation methods using a small set of stable
genes can be used. This approach only requires measurements of genes in the
signature and a small set of stable genes. This approach of scoring can be
invoked by producing a rank matrix by passing in the <code>stableGenes</code> argument
of <code>rankGenes</code>. Stable genes in solid cancers and in blood can be retrieved
using <code>getStableGenes</code>. Upon providing a set of stable genes, <code>rankGenes</code>
automatically ranks all genes relative to these stable genes. When
<code>simpleScore</code> is provided with a rank matrix constructed using stable genes,
it automatically computes scores using a new approach. Details of the set of
stable genes, the new rank estimation approach and the new scoring approach
will soon be published (manuscript in preparation).</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Foroutan, M., Bhuva, D. D., Lyu, R., Horan, K., Cursons, J., &amp;
Davis, M. J. (2018). Single sample scoring of molecular phenotypes. BMC
bioinformatics, 19(1), 1-10.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='rankGenes.html'>rankGenes</a></code>, <code><a href='getStableGenes.html'>getStableGenes</a></code>,
<code><a href='https://rdrr.io/r/base/rank.html'>rank</a></code>, <code>"GeneSet"</code></p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='no'>ranked</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='rankGenes.html'>rankGenes</a></span>(<span class='no'>toy_expr_se</span>)
<span class='no'>scoredf</span> <span class='kw'>&lt;-</span> <span class='fu'>simpleScore</span>(<span class='no'>ranked</span>, <span class='kw'>upSet</span> <span class='kw'>=</span> <span class='no'>toy_gs_up</span>, <span class='kw'>downSet</span> <span class='kw'>=</span> <span class='no'>toy_gs_dn</span>)
<span class='co'># toy_gs_up is a GeneSet object, alternatively a vector of gene ids may also</span>
<span class='co'># be supplied.</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Ruqian Lyu, Momeneh Foroutan, Dharmesh D. Bhuva.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


